### PR TITLE
Ping property added to the scoreboard for each player.

### PR DESCRIPTION
**Pull request by BavoDevos**
Added a new player property to keep track of player their pings to display this on the scoreboard, currently we refresh the ping every 10 seconds as its not necessary to update this frequently.
image_2024-10-01_210725409

### CODE REVIEW

File: Assets/Scripts/GameManagers/InGameManager.cs [hunk 1]
```diff
@@ -937,6 +945,14 @@ private void Update()
             UpdateCleanCharacters();
             EndTimeLeft -= Time.deltaTime;
             EndTimeLeft = Mathf.Max(EndTimeLeft, 0f);
+
+            timeSinceLastPingUpdate += Time.deltaTime;
+            if (timeSinceLastPingUpdate >= pingUpdateInterval)
+            {
+                int currentPing = PhotonNetwork.GetPing();
+                PhotonNetwork.LocalPlayer.SetCustomProperty("Ping", currentPing);
+                timeSinceLastPingUpdate= 0f;
```
Comment (AutumnThyme) [inline Assets/Scripts/GameManagers/InGameManager.cs:954]: fix space missing here

File: Assets/Scripts/CustomLogic/Builtin/CustomLogicPlayerBuiltin.cs [hunk 2]
```diff
@@ -111,6 +113,8 @@ public override void SetField(string name, object value)
                 Player.SetCustomProperty(PlayerProperty.HighestDamage, (int)value);
             else if (name == "TotalDamage")
                 Player.SetCustomProperty(PlayerProperty.TotalDamage, (int)value);
+            else if (name == "Ping")
+                Player.SetCustomProperty(PlayerProperty.Ping, (int)value);
```
Comment (AutumnThyme) [inline Assets/Scripts/CustomLogic/Builtin/CustomLogicPlayerBuiltin.cs:117]: Don't let CL override ping property.

File: Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs [hunk 3]
```diff
@@ -74,8 +74,16 @@ private void SetHeader(ElementStyle style, int currentPlayers, int maxPlayers)
...
+                ElementFactory.CreateDefaultLabel(_header, style, UIManager.GetLocale("ScoreboardPopup", "Scoreboard", "Ping"), FontStyle.Bold, TextAnchor.MiddleCenter);
+
+                _header.GetChild(0).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3;
+                _header.GetChild(1).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3;
+
+                _header.GetChild(2).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 6;
+                _header.GetChild(3).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 6;
+
+                //foreach (Transform t in _header)
+                //    t.GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3f;
```
Comment (AutumnThyme) [inline Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs:86]: Remove unused code.

File: Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs [hunk 4]
```diff
@@ -100,8 +108,17 @@ private Transform CreateRow(ElementStyle style, int index)
...
+            ElementFactory.CreateDefaultLabel(row, style, "0", FontStyle.Normal, TextAnchor.MiddleCenter);
+
+            row.GetChild(0).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3;
+            row.GetChild(1).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3;
+
+            row.GetChild(2).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 6;
+            row.GetChild(3).GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 6;
+
+            //foreach (Transform t in row)
+            //    t.GetComponent<LayoutElement>().preferredWidth = GetPanelWidth() / 3f;
```
Comment (AutumnThyme) [inline Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs:121]: Remove unused code.

File: Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs [hunk 5]
```diff
@@ -126,7 +143,7 @@ private void SetRow(Transform row, Player player)
             else
             {
                 List<string> scoreList = new List<string>();
-                foreach (string property in new string[] { "Kills", "Deaths", "HighestDamage", "TotalDamage" })
+                foreach (string property in new string[] { "Kills", "Deaths", "HighestDamage", "TotalDamage"})
```
Comment (AutumnThyme) [inline Assets/Scripts/UI/InGameMenu/ScoreboardScorePanel.cs:146]: weird spacing

**Review by AutumnThyme — CHANGES_REQUESTED** [review]
See requested changes

**Comment by BavoDevos** [comment]
Changes have been made and this pull request should now be ready to be closed, if no other issues popup ofcours!

**Review by rc174945 — APPROVED** [review]

**Pull request merged**